docs/current_state_specification.md

```diff
@@ -0,0 +1,153 @@
+| type | string | Type of the software node. The supported types are listed below |
+| **Parameter properties** | | |
+| key | string | Key of the parameter |
+| value | string | Value of the parameter |
+| **Association parameter** | | |
+| sourceId | string | Identifier of the source node of the association |
+| targetId | string | Identifier of the target node of the association |
+
+### Supported software types
+
+The list of the supported software types :
+
+| Type | Description |
+| - | - |
+| IMAGE | Represents an image software type |
+| RAW | Represents a raw bytes software type |
+| DATA | Represents a data software type |
+| APPLICATION | Represents an application software type |
+| CONTAINER | Represents a container software type |
+
+### Device Inventory Graph Representation
+
+The diagram below represents the Device Inventory graph and the links between the software and hardware nodes.
+
+The software nodes are organized in the graph at different levels. At the root level stands the main Update Manager software node, which represents the Update Manager component. At the second level are placed the root nodes for each supported domain agent, which are linked with associations to the Update Manager node. At the last level in the hierarchy are placed the domain-specific software nodes, representing the domain components. These software nodes can be modeled the in a tree-based structure if the internal domain-specific representation is more complex.
+
+The hardware nodes do not follow any strict hierarchy and can be linked to any hardware or software node. Any cycles between the nodes are not allowed and prevented.
+
+![Device inventory](/_assets/device-inventory.svg)
+
+### Current State Data Model Example
+
+The following data structure is a holistic example view of a device current state:
+```json
+{
+	"hardwareNodes": [
+		{
+			"id": "cOffee",
+			"version": "rev2",
+			"name": "OWASYS box",
+			"parameters": [
+				{
+					"key": "cpu-arch",
+					"value": "armv7"
+				}
+			]
+		}
+	],
+	"softwareNodes": [
+		{
+			"id": "update-manager",
+			"version": "1.0.0",
+			"name": "Update Manager",
+			"type": "APPLICATION"
+		},
+		{
+			"id": "containers-update-agent",
+			"version": "1.0",
+			"name": "Containers Update Agent",
+			"type": "APPLICATION",
+			"parameters": [
+				{
+					"key": "domain",
+					"value": "containers"
+				},
+				{
+					"key": "container_registry",
+					"value": "ghcr.io"
+				}
+			]
+		},
+		{
+			"id": "containers:helloworld",
+			"version": "1.0.0",
+			"name": "Hello World Example",
+			"type": "CONTAINER"
+		},
+		{
+			"id": "containers:influxdb",
+			"version": "2.7.1",
+			"name": "InfluxDB",
+			"type": "CONTAINER"
+		},
+		{
+			"id": "self-update-agent",
+			"version": "0.2.0",
+			"name": "Self Update Agent",
+			"type": "APPLICATION",
+			"parameters": [
+				{
+					"key": "domain",
+					"value": "self-update"
+				}
+			]
+		},
+		{
+			"id": "self-update:leda-deviceimage",

```

Comment:
Maybe more a question to the self-updater and not the specification. But how would one know where to retrieve this image.

Reply:
Yes, it is a specific question for the self-update agent.
But my understanding is that the ID shall be somewhat constant, e.g. os-image. 
And any extra information for identification of the installed software shall be inside key-value pairs inside the IMAGE software node.